@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -494,9 +494,16 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                     }
                 }
             } else if (isLabelRejectionLineTechnique(technique)) {
+                const point = new THREE.Vector3();
                 for (const path of lines) {
+                    const worldPath: number[] = [];
+                    for (let i = 0; i < path.length; i += 3) {
+                        point.set(path[i], path[i + 1], path[i + 2]);
+                        point.add(this.m_decodeInfo.center);
+                        worldPath.push(point.x, point.y, point.z);
+                    }
                     this.m_linePathGeometries.push({
-                        path
+                        path: worldPath
```

Comment:
Why do we flatten this array here and expand it again in the TileGeometryCreator? When the data is received the array will not contain proper THREE.Vector3 objects but Vector3Like but I guess that's enough?!

Reply:
This is in the decoder, so we need to flatten it to pass it back to the main thread? Or do I miss something.

Reply:
There is no need to flatten it. If you pass an `THREE.Vector[]` from the worker to the main thread it would arrive in the main thread as `Vector3Like[]`(i.e. the methods are lost). Not sure if your code is relying on having full `THREE.Vector3` instances though.

Reply:
cool, I did not know, I will give it a try.

Reply:
it works well, thanks for the hint.

Reply:
> it works well, thanks for the hint.

Well, please consider that here:
```
export interface PathGeometry {
    path: THREE.Vector3[];
}
```
you're lying to Typescript compiler. What you get in main thread is only `Vector3Like` as mentioned by Nino.
To be on safe side, i would change typings in all `PathBlockingElement` related classed and interfaces to `Vector3Like` (it should work) and we're on safe side. 

>  Not sure if your code is relying on having full THREE.Vector3 instances though.

Not yet, but who knows how it will evolve. Unsafe silent conversion like we have in patch now will result in obscure runtime error, when it should be found by compiler.

Reply:
Your right, I will fix.

Reply:
@zbigg, done, please take a look at the latest commit.